lib/puppet/defaults.rb

```diff
@@ -985,31 +985,43 @@ def self.valid_file_checksum_types
       :owner => "service",
       :group => "service",
       :mode => "0755",
-      :desc => "The root directory for the certificate authority."
```

Comment:
I'm really unclear here what the best practice is. I see that other deprecated settings have their description updated. Some issue a deprecation warning in a hook and a couple have a key called `:deprecated` with a value of `:completely` (See `trusted_server_facts`).

Reply:
Yeah I was wondering the same :/ I think for sure updating the description would be good (I put this up mostly to run the tests on it, was planning to add that). But I have no idea what the `:deprecated` key is used for. I'll look into it.

Reply:
I got the warning in first because that seemed the most important to me for surfacing it to people

Reply:
Descriptions updated.

Reply:
Using :deprecated and :completely is preferred. Issuing deprecation warnings in hooks predates the new way. Or if only a certain value is deprecated.

Reply:
Good to know. I need to update this to get all the other CA settings anyway. Thanks Josh!